obspy/taup/seismic_phase.py

```diff
@@ -714,6 +714,48 @@ def parse_name(self, tau_model):
                         "Phase not recognized: {} followed by {}".format(
                             current_leg, next_leg))
 
+            elif current_leg == "Kdiff":
+                # inner core - outer core boundary diffraction
+                if (self.max_ray_param >= tau_model.get_tau_branch(
+                        tau_model.iocb_branch - 1,
+                        is_p_wave).min_turn_ray_param >=
+                        self.min_ray_param):
+                    if (self.current_branch < tau_model.iocb_branch - 1 or
+                        (self.current_branch == tau_model.iocb_branch - 1
+                            and end_action != _ACTIONS["diffract"])):
```

Comment:
In principle, I think in all these cases [my comment in the other ticket](https://github.com/obspy/obspy/pull/3080#discussion_r902769093) applies in the same way. Readability could be much improved by using named variables for the individual parts of the conditions. But I don't want to hassle you too much with it, if you feel it's OK to do it this way (I assume it might be like this throughout the taup code base..?)

Reply:
I agree that the readibility could be improved, but I'm afraid I'm not inclined to do so for this PR. What I've done here done is consistent with the existing style (e.g. see how Pdiff and Sdiff are treated in lines 555-604). Really there should be some refactoring and cleaning up of the whole complicated `SeismicPhase.parse_name` method, and this should be done consistently e.g. whatever is done for Kdiff should also be done for Pdiff. But I think that's a big job, and requires careful thought.

I'm fine with the cosmetic change you made.